sdks/python/apache_beam/testing/load_tests/load_test_metrics_utils.py

```diff
@@ -138,8 +143,25 @@ def as_dict(self):
 class CounterMetric(Metric):
   def __init__(self, counter_dict, submit_timestamp, metric_id):
     super(CounterMetric, self).__init__(submit_timestamp, metric_id)
-    self.value = counter_dict.committed
     self.label = str(counter_dict.key.metric.name)
+    self.value = counter_dict.committed
+
+
+class DistributionMetrics(Metric):
```

Comment:
You are right, naming here isn't very accurate. To give more context:
Metrics such as counters or the runtime metric are just one value. Runtime is evaluated from multiple distribution metrics, but distribution metric has 4 values inside (sum, mean, max, min). There was situation when metrics weren't introduced exactly in test but in the pipelines in project (it happened in tfx project) and I wasn't sure what should be saved in those cases, so I decided that we should save everything.
Also it's not a dumb question :)

Reply:
[Here](https://github.com/apache/beam/pull/8731) was decision about saving all metrics (not only specified by user in test).

Reply:
Have you considered saving median instead of mean? a single outlier may be affecting mean for the time window it is considered. Evaluating mean (aka 50th percentile) would produce a more reliable signal. One can also consider other percentiles, p90, p99 if necessary.

Reply:
How do you capture the time window over which you collected an aggregate metric (min/max)?

Reply:
I understand your point, calculating one value might "save" some space in BigQuery. Unfortunately distribution metrics doesn't support with median or other percentiles out of the box (maybe it's a good idea for feature request?) and I am not sure if it's good idea to introduce such logic in util for reading metrics. Mainly because in the situation when we would like save some external (not expected in MetricsReader) metrics which we don't know what mean (but still may be valuable for the others ex. tfx team) calculating median or any aggregates may cause loosing meaningful info (maybe someone who defined those external metrics needs max, min, sum or mean). Do you agree?

As it comes to time windows, also distribution doesn't store such thing. It is possible to retrieve the step name in which metrics would be collected. And this I can add to metric label we save.

This is confusing problem. To sum everything up:
- What to do with metrics that weren't collected in our pipeline but somewhere deeper? It is decided to save them. But I can suggest here that it would help to have pipeline option called `save_external_metrics`. WDYT?
- In what shape we should save metrics that we don't know nothing about them? IMO probably as raw as possible, because we don't know what is useful for user that collects them.

Reply:
I see, looks like saving median would require changes to beam. Regarding unknown distributions, I see following options:
1) Save sum, count, min, max, collected across all distributions associated with a metric with the same name.  In other words, total sum, total count, min of minimums and max of maximums. 
2) Save `(sum, count, min, max)` for each distribution without aggregation (not sure if useful, but since we are exporting to bigquery, we can run a query later to aggregate something we need).
3) register aggregators with an instance of MetricsReader. Registration will take the name of the metric, and an aggregator instance that recieves a list of all distribution readings, and computes desired output to store into Bigquery. 

I think 2 is most flexible, but perhaps 1) is sufficient and will cover most of the use-cases. Feel free to ask @robertwb or @pabloem for an opinion. 3 seems like overengineering at this point.

Reply:
I am okay with either 1 or 2. Do we have full metric names( step+namespace+metric)?

Reply:
Ok, so I changed it into option 2. I changes it also to have name namespace+step(where possible)+metric.